owncloudData/src/main/java/com/owncloud/android/data/capabilities/datasources/implementation/OCLocalCapabilitiesDataSource.kt

```diff
@@ -19,29 +19,95 @@
+        ocCapabilityDao.getCapabilitiesForAccount(accountName)?.toModel()
 
     override fun insert(ocCapabilities: List<OCCapability>) {
         ocCapabilityDao.replace(
-            ocCapabilities.map { ocCapability -> ocCapabilityMapper.toEntity(ocCapability)!! }
+            ocCapabilities.map { ocCapability -> ocCapability.toEntity() }
         )
     }
+
+    companion object {
+        @VisibleForTesting
+        fun OCCapabilityEntity.toModel(): OCCapability =
+            OCCapability(
+                id = id,
+                accountName = accountName,
+                versionMayor = versionMayor,
+                versionMinor = versionMinor,
+                versionMicro = versionMicro,
+                versionString = versionString,
+                versionEdition = versionEdition,
+                corePollInterval = corePollInterval,
+                davChunkingVersion = davChunkingVersion,
+                filesSharingApiEnabled = CapabilityBooleanType.fromValue(filesSharingApiEnabled),
+                filesSharingPublicEnabled = CapabilityBooleanType.fromValue(filesSharingPublicEnabled),
+                filesSharingPublicPasswordEnforced = CapabilityBooleanType.fromValue(filesSharingPublicPasswordEnforced),
+                filesSharingPublicPasswordEnforcedReadOnly = CapabilityBooleanType.fromValue(filesSharingPublicPasswordEnforcedReadOnly),
+                filesSharingPublicPasswordEnforcedReadWrite = CapabilityBooleanType.fromValue(filesSharingPublicPasswordEnforcedReadWrite),
+                filesSharingPublicPasswordEnforcedUploadOnly = CapabilityBooleanType.fromValue(filesSharingPublicPasswordEnforcedUploadOnly),
+                filesSharingPublicExpireDateEnabled = CapabilityBooleanType.fromValue(filesSharingPublicExpireDateEnabled),
+                filesSharingPublicExpireDateDays = filesSharingPublicExpireDateDays,
+                filesSharingPublicExpireDateEnforced = CapabilityBooleanType.fromValue(filesSharingPublicExpireDateEnforced),
+                filesSharingPublicUpload = CapabilityBooleanType.fromValue(filesSharingPublicUpload),
+                filesSharingPublicMultiple = CapabilityBooleanType.fromValue(filesSharingPublicMultiple),
+                filesSharingPublicSupportsUploadOnly = CapabilityBooleanType.fromValue(filesSharingPublicSupportsUploadOnly),
+                filesSharingResharing = CapabilityBooleanType.fromValue(filesSharingResharing),
+                filesSharingFederationOutgoing = CapabilityBooleanType.fromValue(filesSharingFederationOutgoing),
+                filesSharingFederationIncoming = CapabilityBooleanType.fromValue(filesSharingFederationIncoming),
+                filesBigFileChunking = CapabilityBooleanType.fromValue(filesBigFileChunking),
+                filesUndelete = CapabilityBooleanType.fromValue(filesUndelete),
+                filesVersioning = CapabilityBooleanType.fromValue(filesVersioning)
```

Comment:
We have a bunch of properties here, maybe splitting them up in some new classes? 🍻

Reply:
In this PR, I just wanted to remove the mappers. Splitting this class into smaller ones will need probably database changes and migrations and so on. So, I would keep it at least for the moment 👍